### PR TITLE
[4.0] html5 doesnt need slashes for selfclosing

### PR DESCRIPTION
**Pull request by brianteeman**
continue the removal of closing slashes for tags that do not require this in html5

hr
meta
link
br

Please note this PR does not include "required?" changes to the system language code plugin as I am not 100% sure I understand the regex there. - that can be done either as a new PR or as a pull request against this one.

@wilsonge @C-Lodder this is a redo of  #15877

### CODE REVIEW

**Comment by dgt41** [comment]
@brianteeman may I suggest another change?
Instead of  using a very generic HR tag with an id `<hr id="system-readmore">`
we can have something more specific for the task, e.g. `<joomla-article-readmore></joomla-article-redmore>`

**Comment by brianteeman** [comment]
@dgt41 that is beyond the scope of this PR and potentially introduces an unneeded bc issue

File: tests/unit/suites/libraries/joomla/document/feed/renderer/JDocumentRendererAtomTest.php [hunk 1]
```diff
@@ -116,13 +116,13 @@ public function testRender()
 <feed xmlns="http://www.w3.org/2005/Atom"  xml:lang="en-gb">
 	<title type="text"></title>
 	<subtitle type="text"></subtitle>
-	<link rel="alternate" type="text/html" href="http://localhost"/>
```
Comment (wilsonge) [inline tests/unit/suites/libraries/joomla/document/feed/renderer/JDocumentRendererAtomTest.php]: This one isn't right. This is a genuine XML document. We can't skip close slashes here

File: tests/unit/suites/libraries/joomla/feed/parser/JFeedParserAtomTest.php [hunk 2]
```diff
@@ -136,7 +136,7 @@ public function testHandleId()
 	public function testHandleLink()
 	{
 		// Setup the inputs.
-		$el   = new SimpleXMLElement('<link href="http://domain.com/" />');
```
Comment (wilsonge) [inline tests/unit/suites/libraries/joomla/feed/parser/JFeedParserAtomTest.php]: And here

File: tests/unit/suites/libraries/joomla/feed/parser/JFeedParserRssTest.php [hunk 3]
```diff
@@ -275,7 +275,7 @@ public function testHandleLastBuildDate()
 	public function testHandleLink()
 	{
 		// Setup the inputs.
-		$el   = new SimpleXMLElement('<link href="http://domain.com/path/to/resource" />');
```
Comment (wilsonge) [inline tests/unit/suites/libraries/joomla/feed/parser/JFeedParserRssTest.php]: And here

File: tests/unit/suites/libraries/joomla/feed/parser/JFeedParserAtomTest.php (hunk 2 above)
Comment (brianteeman) [inline tests/unit/suites/libraries/joomla/feed/parser/JFeedParserAtomTest.php]: reverted

File: tests/unit/suites/libraries/joomla/document/feed/renderer/JDocumentRendererAtomTest.php (hunk 1 above)
Comment (brianteeman) [inline tests/unit/suites/libraries/joomla/document/feed/renderer/JDocumentRendererAtomTest.php]: reverted

File: tests/unit/suites/libraries/joomla/feed/parser/JFeedParserRssTest.php (hunk 3 above)
Comment (brianteeman) [inline tests/unit/suites/libraries/joomla/feed/parser/JFeedParserRssTest.php]: reverted

**Comment by wilsonge** [comment]
Awesome - thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**